components/movementsensor/gpsutils/ntrip.go

```diff
@@ -162,78 +112,37 @@ Loop:
 				logger.Debug("Reached the end of SourceTable")
 				break Loop
 			}
-			return nil, fmt.Errorf("%s: illegal sourcetable line: '%s'", n.URL, ln)
+			return nil, fmt.Errorf("%s: illegal sourcetable line: '%s'", n.url, ln)
 		}
 	}
 
 	return st, nil
 }
 
-// ParseStream parses a line from the sourcetable.
-func parseStream(line string) (Stream, error) {
-	fields := strings.Split(line, ";")
-
-	// Standard stream contains 19 fields.
-	// They are enumerated by their constants at the top of the file
-	if len(fields) < 19 {
-		return Stream{}, fmt.Errorf("missing fields at stream line: %s", line)
-	}
-
-	if fields[carrierField] == "" {
-		fields[carrierField] = "0"
-	}
-	carrier, err := strconv.Atoi(fields[carrierField])
-	if err != nil {
-		return Stream{}, fmt.Errorf("cannot parse the streams carrier in line: %s", line)
-	}
-
-	satSystems := strings.Split(fields[navsystem], "+")
-
-	lat, err := strconv.ParseFloat(fields[latitude], floatbitsize)
-	if err != nil {
-		return Stream{}, fmt.Errorf("cannot parse the streams latitude in line: %s", line)
-	}
-	lon, err := strconv.ParseFloat(fields[longitude], floatbitsize)
-	if err != nil {
-		return Stream{}, fmt.Errorf("cannot parse the streams longitude in line: %s", line)
-	}
-
-	nmea, err := strconv.ParseBool(fields[nmeaBit])
-	if err != nil {
-		return Stream{}, fmt.Errorf("cannot parse the streams nmea in line: %s", line)
-	}
-
-	sol, err := strconv.Atoi(fields[solution])
+// Connect attempts to initialize a new ntrip client. If we're unable to connect after multiple
+// attempts, we return the last error.
+func (n *NtripInfo) Connect(ctx context.Context, logger logging.Logger) error {
+	err := n.createConnection(ctx, logger)
 	if err != nil {
-		return Stream{}, fmt.Errorf("cannot parse the streams solution in line: %s", line)
-	}
-
-	fee := false
-	if fields[feeBit] == "Y" {
-		fee = true
+		return err
 	}
 
-	bitrate, err := strconv.Atoi(fields[bitRateField])
+	err = n.waitUntilCasterIsLive(logger)
 	if err != nil {
-		bitrate = 0
+		return err
 	}
 
-	return Stream{
-		MP: fields[mp], Identifier: fields[id], Format: fields[format], FormatDetails: fields[formatDetails],
-		Carrier: carrier, NavSystem: satSystems, Network: fields[network], Country: fields[country],
-		Latitude: float32(lat), Longitude: float32(lon), Nmea: nmea, Solution: sol, Generator: fields[generator],
-		Compression: fields[compression], Authentication: fields[auth], Fee: fee, BitRate: bitrate, Misc: fields[misc],
-	}, nil
+	return nil
```

Comment:
do we need to close the connection if this fails?

if not, then `return n.waitUntilCasterIsLive(logger)`?

Reply:
ooh, good question! We talked in person: we _don't_ need to close the connection if it fails because `waitUntilCasterIsAlive` cleans up after itself, and `createConnection` does not actually make a long-term connection to the caster (that happens in `getStream`, and perhaps suggests that some of these functions are misnamed. but finding better names can wait for a different PR). So, I should just clean up the return statements.